src/components/ToggleButton/ToggleButton.js

```diff
@@ -1,134 +1,109 @@
+      <FontAwesomeIcon
+        icon={['fas', 'list']}
+        className="slider-icon--grid-list"
+      />
+      <Switch {...props} className={classNames(className, 'slider--teal')} />
+      <FontAwesomeIcon
+        icon={['fas', 'th']}
+        className="slider-icon--grid-list"
+      />
+    </div>
+  );
+}
 
-    return (
-      <label className="switch my-auto mx-1">
-        <input
-          type="checkbox"
-          checked={value}
-          onChange={this.handleClick}
-          name={title}
-        />
-        <span className={joinedClasses}></span>
-      </label>
-    );
-  }
+function ButtonElement(props) {
+  const { buttonProps, className, iconProps, title, icon, onClick } = props;
+  const classes = classNames(className, 'btn-toggle');
+
+  return (
+    <Button
+      {...buttonProps}
+      data-testid="toggle-button"
+      title={title}
+      onClick={onClick}
+      className={classes}
+    >
+      <FontAwesomeIcon size="lg" {...iconProps} icon={icon} />
+    </Button>
+  );
+}
 
-  get gridList() {
-    return (
-      <div className="slider--grid-list">
-        <FontAwesomeIcon
-          icon={['fas', 'list']}
-          className="slider-icon--grid-list"
-        />
-        {this.switch('slider--teal')}
-        <FontAwesomeIcon
-          icon={['fas', 'th']}
-          className="slider-icon--grid-list"
-        />
-      </div>
-    );
-  }
+export default function ToggleButton(props) {
+  const { iconOnly, variant, value, onClick } = props;
 
-  get button() {
-    const { buttonProps, className, iconProps, title } = this.props;
-    const classes = classNames(className, 'btn-toggle');
+  const derivedProps = {
+    ...props,
+    icon: value ? ['fas', 'check-square'] : ['far', 'square'],
+    onClick: (event) => onClick && onClick(event)
+  };
 
-    return (
-      <Button
-        {...buttonProps}
-        data-testid="toggle-button"
-        title={title}
-        onClick={this.handleClick}
-        className={classes}
-      >
-        <FontAwesomeIcon size="lg" {...iconProps} icon={this.icon} />
-      </Button>
-    );
+  if (iconOnly) {
+    return <IconOnly {...derivedProps} />;
   }
 
-  get iconOnly() {
-    const { className, iconProps, title } = this.props;
-    const classes = classNames(className, 'icon-toggle');
-
-    return (
-      <FontAwesomeIcon
-        size="lg"
-        {...iconProps}
-        icon={this.icon}
-        title={title}
-        onClick={this.handleClick}
-        className={classes}
-      />
-    );
+  switch (variant) {
+    case 'switch':
+      return <Switch {...derivedProps} />;
+    case 'grid-list':
+      return <GridList {...derivedProps} />;
+    default:
+      return <ButtonElement {...derivedProps} />;
   }
+}
 
-  render() {
-    const { iconOnly, variant } = this.props;
-
-    if (iconOnly) {
-      return this.iconOnly;
-    }
+IconOnly.propTypes = Switch.propTypes = GridList.propTypes = ButtonElement.propTypes = ToggleButton.propTypes = {
```

Comment:
```[IconOnly, Switch, GridList, ButtonElement, ToggleButton].map(
  (Comp) =>
    (Comp.propTypes = {
      className: PropTypes.string,
      buttonProps: PropTypes.object,
      iconProps: PropTypes.object,
      title: PropTypes.string.isRequired,
      value: PropTypes.bool,
      iconOnly: PropTypes.bool,
      variant: PropTypes.oneOf(toggleButtonProps.variants),
      onClick: PropTypes.func
    })
);```

I investigated doing the above to remove the redundant `.propTypes =`, but it doesn't work properly with eslint

Reply:
Oh, just declare the prop types as a constant and then assign the constant to each component's propTypes

```
const buttonProps = {
  className: PropTypes.string,
   ...
};

Icon.propTypes = buttonProps;
GridList.propTypes = buttonProps;
...
```